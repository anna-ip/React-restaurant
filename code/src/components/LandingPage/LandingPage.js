import React from 'react'
import Links from '../Links'
import Reviews from './ReviewContainer/Reviews'
import styles from './LandingPage.module.scss'
import mainImage from '../../images/pexels-emre-can-2079438.jpg'

const LandingPage = () => {
  return (
    <div className={styles.landingpageContainer}>
      <div className={styles.linkContainer}>
        <Links to={'/menu'} text={'Take me to the'} />
        <Links to={'/menu'} text={'Menu'} />
      </div>
      <img src={mainImage} alt="restaurant" className={styles.img} />
      <Reviews />
    </div>
  )
}

export default LandingPage
