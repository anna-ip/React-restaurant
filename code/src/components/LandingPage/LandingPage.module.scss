@import '../../styles/_variables.scss';
@import '../../styles/mixins';

.landingpageContainer {
  @include flexCenter;
}
.linkContainer {
  @include flexCenter;
  justify-content: center;
  height: 5rem;
}
.img {
  width: 90%;
  height: auto;
  border: $image-border;
}
