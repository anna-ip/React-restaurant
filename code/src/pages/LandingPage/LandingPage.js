import React from 'react'
import Reviews from '../../components/ReviewComponents/REviewComponents/Reviews/Reviews'
import styles from './LandingPage.module.scss'
import mainImage from '../../images/pexels-emre-can-2079438.jpg'

const LandingPage = () => {
  return (
    <div className={styles.landingPageContainer}>
      <img src={mainImage} alt="restaurant" className={styles.img} />
      <Reviews />
    </div>
  )
}

export default LandingPage
