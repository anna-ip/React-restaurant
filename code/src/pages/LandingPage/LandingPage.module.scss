@import '../../styles/_variables.scss';
@import '../../styles/mixins';

.landingPageContainer {
  @include flexCenter;
}

.img {
  width: 90%;
  height: auto;
  @include imageBorder;
}
