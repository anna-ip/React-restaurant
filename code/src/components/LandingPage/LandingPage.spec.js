import React from 'react'
//import renderer from 'react-test-renderer'
import LandingPage from './LandingPage'

import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
configure({ adapter: new Adapter() })

test('<LandingPage/> renders correctly using enzyme', () => {
  const tree = shallow(<LandingPage />)
  expect(toJson(tree)).toMatchSnapshot()
})

// test('Render correctly', () => {
//   // TypeError: Cannot read property 'reviews' of undefined
//   // const tree = renderer.create(<LandingPage />).toJSON()
//   // console.log(tree)
//   // expect(tree).toMatchSnapshot()
// })
