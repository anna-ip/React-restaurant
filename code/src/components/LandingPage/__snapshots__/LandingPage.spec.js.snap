// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LandingPage/> renders correctly using enzyme 1`] = `
<div
  className="landingpageContainer"
>
  <div
    className="linkContainer"
  >
    <Links
      text="Take me to the"
      to="/menu"
    />
    <Links
      text="Menu"
      to="/menu"
    />
  </div>
  <img
    alt="restaurant"
    className="img"
    src="pexels-emre-can-2079438.jpg"
  />
  <Reviews />
</div>
`;
